modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesContextStore.java

```diff
@@ -77,9 +77,17 @@ void storeDurableSubscription(AndesSubscription subscription)
      *
      * @param subscription The subscriptoin object to update
      * @throws AndesException
+     * @return If a subscription was updated return 1, else return 0
      */
-    void updateDurableSubscription(AndesSubscription subscription)
-		    throws AndesException;
+    int updateDurableSubscription(AndesSubscription subscription) throws AndesException;
+
+    /**
+     * Update already the store if there is a matching subscription. Else, insert the subscription.
+     *
+     * @param subscription The subscriptoin object to update
+     * @throws AndesException
+     */
+    void updateOrInsertDurableSubscription(AndesSubscription subscription) throws AndesException;
```

Comment:
The priority is to update. that's why I named it as updateOrInsert can rename  if it seems odd. We already have two separate methods, here, we need to insert the subscription only if it cannot be updated(If it's not there)